libkbfs/conflict_resolver.go

```diff
@@ -195,8 +206,16 @@ func (cr *ConflictResolver) Resolve(unmerged MetadataRevision,
 		return
 	}
 
+	cr.inputLock.Lock()
+	defer cr.inputLock.Unlock()
```

Comment:
I think the lock shouldn't be held when sending on `inputChan`, as that could lead to deadlock. Here's the scenario:

main goroutine: `Resolve` is called with rev=N, skips the cancel, gets to the send on `inputChan`.
bg goroutine: `processInput` receives on `inputChan`
main goroutine: `Resolve` is called with rev=N+1, skips the cancel, gets to the send on `inputChan` while `inputLock` is held.
bg goroutine: `processInput` enters the anonymous function, holds `inputLock`.